packages/core/src/document.ts

```diff
@@ -378,17 +378,77 @@ class Document extends EventEmitter {
     return (await this._findIndex(cid, log)) !== -1
   }
 
+  /**
+   * Given two different DocStates representing two different conflicting histories of the same
+   * document, pick which version to accept, in accordance with our conflict resolution strategy
+   * @param state1 - first log's state
+   * @param state2 - second log's state
+   * @returns true if state2's log should be taken, or false if state1's log should be taken
+   * @private
+   */
+  static async _pickLogToAccept(state1: DocState, state2: DocState): Promise<boolean> {
+    const isState1Anchored = state1.anchorStatus === AnchorStatus.ANCHORED
+    const isState2Anchored = state2.anchorStatus === AnchorStatus.ANCHORED
+
+    if (isState1Anchored != isState2Anchored) {
+      // When one of the logs is anchored but not the other, take the one that is anchored
+      return isState2Anchored
+    }
+
+    if (isState1Anchored && isState2Anchored) {
+      // compare anchor proofs if both states are anchored
+      const { anchorProof: proof1 } = state1
+      const { anchorProof: proof2 } = state2
+
+      if (proof1.chainId == proof2.chainId) {
+        // The logs are anchored in the same blockchain, compare block heights to decide which to take
+        if (proof1.blockNumber < proof2.blockNumber) {
+          return false
+        } else if (proof2.blockNumber < proof1.blockNumber) {
+          return true
+        }
+        // If they have the same block number fall through to fallback mechanism
+      } else {
+        // The logs are anchored in different blockchains, compare block timestamps to decide which to take
+        if (proof1.blockTimestamp < proof2.blockTimestamp) {
+          return false
+        } else if (proof2.blockTimestamp < proof1.blockTimestamp) {
+          return true
+        }
+        // If they have the same block timestamp fall through to fallback mechanism
+      }
+    }
+
+    // The anchor status is the same between both logs.  If either log has a 'nonce' that means
+    // multiple updates have been squashed into a single anchor window, so prefer the log with the
+    // higher nonce (which indicates it has had more writes).
+    const nonce1 = (state1.next?.metadata.nonce ?? state1.metadata.nonce) || 0
```

Comment:
This now leaks the concept of a 'nonce' outside of the tile-doctype for the first time.  Longer term I imagine all conflict resolution behavior will move into the specific doctype classes and out of Document.